File: src/main/python/Day6.py
import Utils as Utils

def calculateLanternFish(lines, days):
    for i in range(days):
        amount = lines[0]
        lines = lines[1:]
        lines[6] += amount
        lines.append(amount)
    return sum(lines)

@Utils.time_it_and_evaluate
def partOne(lines):
    return calculateLanternFish(lines.copy(), 80)

@Utils.time_it_and_evaluate
def partTwo(lines):
    return calculateLanternFish(lines.copy(), 256)

if __name__ == '__main__':
    lines = [ int(x) for x in Utils.readAll(2021, 'day6').rstrip().split(',') ]
    countList = [ 0 ] * 9
    for x in lines:
        countList[x] += 1

    print('Day 6')
    partOne(countList)
    partTwo(countList)

File: src/main/java/hardcoded/aoc2021/Day6.java
package hardcoded.aoc2021;

import java.util.stream.LongStream;

import hardcoded.util.Utils;

public class Day6 {
	public static void main(String[] args) throws Exception {
//		int[] days = Arrays.asList(Utils.readAll(2021, "day6").trim().split(",")).stream().mapToInt(Integer::parseInt).toArray();
		int[] days = Utils.readIntArray(2021, "day6", ",");
		
		long[] scores = new long[9];
		for(int i : days) {
			scores[i]++;
		}
		
		Utils.printf("Day 6\n");
		Utils.printf("PartOne: %d\n", partOne(scores));
		Utils.printf("PartTwo: %d\n", partTwo(scores));
	}
	
	public static long calculateLanternFish(long[] array, int days) {
		for(int i = 0; i < days; i++) {
			long amount = array[0];
			
			// Shift all values by one
			System.arraycopy(array, 1, array, 0, array.length - 1);
			
			array[6] += amount;
			array[8] = amount;
		}
		
		return LongStream.of(array).sum();
	}
	
	public static long partOne(long[] array) throws Exception {
		return calculateLanternFish(array.clone(), 80);
	}
	
	public static long partTwo(long[] array) throws Exception {
		return calculateLanternFish(array.clone(), 256);
	}
}
